bootstrap/interfaces/secret.go

```diff
@@ -28,4 +28,10 @@ type SecretProvider interface {
 
 	// HasSecret returns true if the service's SecretStore contains a secret at the specified path.
 	HasSecret(path string) (bool, error)
+
+	// RegisteredSecretUpdateCallback registers a callback for a secret.
+	RegisteredSecretUpdateCallback(path string, callback func(path string))
```

Comment:
Would it be useful for the callback to actually supply the callee with the new secret value?

```suggestion
	RegisteredSecretUpdateCallback(path string, callback func(path string, secrets map[string]string))
```

Reply:
Seems to be that if we have a Register() that we should have an UnRegister() as well.